src/linear_algebra.jl

```diff
@@ -0,0 +1,56 @@
+for (jf, af) in [(:+, :add!), (:-, :sub!), (:*, :mul!)]
+    @eval function Base.$jf(A::T, B::T) where {T<:Union{ArbMatrix,AcbMatrix}}
+        C = T(size(A, 1), size(B, 2); prec = max(precision(A), precision(B)))
+        $af(C, A, B)
+        C
+    end
+end
```

Comment:
We should check that the dimensions agree here and throw an error otherwise. This would be done differently for addition and multiplication.

Reply:
Similar things should be done for some of the other methods.

Reply:
I added bounds check for all the linear algebra routines